---
title: 'METHODOLOGY: Transcriptomic analysis of the black widow spider venom (*Latrodectus curacaviensis*)'
author: "Gabriel Alves Vieira"
date: "April 2020"
header-includes:
   - \usepackage{indentfirst}
   - \newcommand{\codebox}[1] {
     \colorbox{gray!10}{\texttt{#1}}}
output:
   bookdown::word_document2:
      #highlight: espresso
      #reference_docx: template_justif_indent.docx
   bookdown::pdf_document2:
     #fig_caption: yes
     #latex_engine: xelatex
     toc: true
     toc_depth: 2
lof: true
lot: true
indent: true
fontsize: 12pt
urlcolor: blue
citecolor: blue
#geometry:
#- top=2cm
#- left=2cm
#- right=2cm
#- bottom=2cm
#- heightrounded
link-citations: yes
csl: biomed-central.csl
bibliography: library.bib
---


<!-- Adding CSS to change text (interesting if the outfmt is html)  

<style>

#body {
#text-align: justify}

#br {
#content: " " !important;
#display: block !important;
#margin:10px;}

#table, th, td {
#  border: 1px solid black;
#}

</style>
-->

# Transcriptome sequencing and assembly

The Illumina HiSeq 1500 equipment
(Illumina, San Diego, California, United States) was used for RNA sequencing, with ribosomal RNA reduction (rRNA) reagents and large-scale sequencing. After the
sequencing platform generated the sequencing images, the CASAVA software (version 1.8.2) was used to separate samples by identifying the barcodes assigned during the cDNA library preparation step. The final step was the conversion from BCL files to fastq format files that contains only nucleotide sequences and their respective qualities, discarding reads
that have not been approved by the Q score base quality control (greater than phred 30).
The reads were assembled using Trinity software @Grabherr2011 with a minimum contig size of 120 (instead of the default value: 200). Trinity data allowed us to identify the number of
isoforms per gene and other features.

# Bioinformatics analyses

Three different methodologies were used to annotate the contigs generated by Trinity:

1. Alignment using BLASTx to several databases, including: (i) UNIPROT (download on July, the 27 th , 2017 UniprotMetazoaToxin), (ii) eukaryotic COGs, (iii) Arachnoserver @Pineda2018, and (iv) an subset of UNIPROT including only sequences from the clade Arthropoda. The contigs were compared to these databases BLASTx to provide sequence similarity information. Only the hits an e-value inferior to 1e-5 were considered and we used the best hit against each database to provide a putative annotation for each contig. The contigs were also classified according to a ranking of expression as observed by TPM (transcripts per million transcripts) measures provided by Trinity. 

   We also performed a specific ranking containing only the contigs annotated as toxins by Arachnoserver best hits. All mRNA transcripts for the spider genus Latrodectus available at Genbank were downloaded and converted into a BLAST database. Our transcripts were aligned against this database using BLASTn 2.9.0 with an evalue of 10.

2. Submission of sequences to the Trinotate annotation pipeline v3.2.1 @Bryant2017 using the autoTrinotate perl script. Transdecoder (https://github.com/TransDecoder/TransDecoder/wiki), a companion utility for Trinity @Haas2013, was used to extract and translate ORFs from the assembled contigs. Blast 2.9.0 searches were run using the entire transcripts (BlastX) and ORF aminoacid sequences (BlastP) against the Swissprot database @TheUniProtConsortium2019. HMM searches against the PFAM 32.0 database @El-Gebali2018 were performed using Hmmer 3.2.1 (http://hmmer.org/) for protein domain identification. Specific annotation was also performed for signal peptides with signalP 4.0 @Petersen2011, Transmembrane domains with TMHMM 2.0c  @Krogh2001 and rRNAs with RNAmmer 1.2 @Lagesen2007. Lastly, Trinotate provides annotation from GO @GeneOntologyConsortium2004, eggNOG @Jensen2008 and KEGG @Kanehisa2000 databases to contigs. Results were integrated into a sqlite database, used to generate a .xls tab-separated report.

   Trinotate was run with the default parameters specified in autoTrinotate.pl configuration file, except for Transdecoder, where the `-m` parameter was used to extract ORFs longer than 30 aminoacids from our assembled contigs:

   `TransDecoder.LongOrfs -t nt.fasta -m 30 > protein.fasta`
   
     This custom parameter was necessary because by default Transdecoder will only keep ORFs longer than 100 aminoacids, and some proteins of interest for this study are below this threshold. Inhibitor cystine-knots (ICKs), for instance, are generally 30 to 50 residues long @Postic2018.

<!--
#   \begin{center}
#   \codebox{TransDecoder.LongOrfs -t nt.fasta -m 30 > protein.fasta}
#   \end{center}
-->

3. After running autoTrinotate, we have also performed a custom Blast searches search against mRNA sequences from *Latrodectus* spp. available at Genbank. All *Latrodectus* mrna sequences were downloaded in November 2019 and cleaned with CD-HIT 4.8.1 @Fu2012. The `-c 1` parameter was used to ensure that only duplicate sequences were removed. 

<!--
#   \begin{center}
#   \codebox{cd-hit -c 1 ...}
#   \end{center}

   The difference in sequences is evidenced in Table \@ref(tab:mrna)

   ```{r mrna, echo=FALSE}
   mrna_latrod <- data.frame("Raw"=14067, "Cleaned"=14013)
   knitr::kable(mrna_latrod,
             caption = "Number of sequences before and after cleanup")
   ```
   
   You can see *Latrodectus curacaviensis* morphology in Figure \@ref(fig:cura)
   
   
   ```{r cura, fig.cap = "\\textit{L. curacaviensis} morphology", out.width = "100%"}
   knitr::include_graphics("cura.jpg")
   ```
-->

   Then, ORFs were extracted from the datasets using Transdecoder with the same `-m 30` specification used in Trinotate.


# REFERENCES